projects/vdk-core/src/vdk/internal/builtin_plugins/run/data_job.py

```diff
@@ -54,6 +54,32 @@ def new_datajob(
         return DataJob(data_job_directory, core_context, name)
 
 
+def save_step_result(context: JobContext, result: StepResult):
```

Comment:
I think we should encapsulate things in a class . there's a lot of "knowledge" require to use this effectively

Reply:
The two functions are only meant to be used inside that file. I can make them class members of DataJob, it'll be less likely someone imports them in another file.